src/Features/Authentication/AccessControlFilter.cs

```diff
@@ -0,0 +1,41 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Aptabase.Data;
+using System;
+
+namespace Aptabase.Features.Authentication
+{
+    public class AccessControlFilter : ActionFilterAttribute
```

Comment:
Rename this to `HasAccessToAppFilter`

Reply:
extend System.Web.Http.Filters.ActionFilterAttribute instead, and use the async overload of OnActionExecuting